src/result_collection.jl

```diff
@@ -0,0 +1,109 @@
+using DataFrames
+using BSON
+
+"""
+    merge_dataframes(df, df_new)
+Merge two dataframes `df` and `df_new`. If the `names` of the dataframes
+are the same this is just `vcat`.
+
+If `df_new` is missing keys that are already columns in `df`,
+they will set as `missing` in `df`.
+If on the other hand new keys are encountered, existing in `df_new`
+but not `df`, a new column will be added and filled with `missing`
+in `df`. Then `df` and `df_new` are concatenated.
+"""
+function merge_dataframes(df1, df2)
+    if names(df1) == names(df2)
+        return vcat(df1, df2)
+    else
+        for m ∈ setdiff(names(df1), names(df2))
+            df2[m] = [missing]
+        end
+        for m ∈ setdiff(names(df2), names(df1))
+            df1[m] = fill(missing, size(df1,1))
+        end
+        return vcat(df1,df2)
+    end
+end
+
+is_valid_file(file, valid_filetypes) =
+    any(endswith.(Ref(file),valid_filetypes))
+
+function to_data_row(data,
+        white_list = collect(keys(data)),
+        black_list = [],
+        special_list = [])
+    cnames = setdiff!(white_list, black_list)
```

Comment:
I think this kind of `setdiff!` processing should happen once in the main function instead of being repeated everytime in `to_data_row`.

Reply:
Ah now I see. this cannot be done, because you are taking into acount already the possibility of different datasets with different keys each. Very good!